pepper/apps/dex_common.h

```diff
@@ -1,15 +1,17 @@
+#pragma once
+
 #ifdef PEPPER
 // Pepper (C) doesn't know bool
 typedef unsigned char bool;
 #endif
 
 /**
- * Copies `lenght` bits from source to target
+ * Copies `length` bits from source to target
  */
-void copyBits(bool* source, bool* target, uint32_t length) {
+void copyBits(bool* source, uint32_t source_offset, bool* target, uint32_t target_offset, uint32_t length) {
```

Comment:
This change had to be made in order to avoid constraint explosion. If instead of passing in the offset, I pass in the manipulated pointer directly, e.g instead of
```
copyBits(in, index, pedersenInput, 254+padding, chunkSize);
copyBits(in + index, pedersenInput+254+padding, chunkSize);
```

The constraint size explodes from 10k to 200k. Interestingly this also only happens when we extract this logic into its own method. If done in the main compute method directly, it doesn't lead to constraint explosion.

😕

Reply:
okay, strage....